spark/client/base/base.go

```diff
@@ -48,5 +48,6 @@ type SparkConnectClient interface {
 
 type ExecuteResponseStream interface {
 	ToTable() (*types.StructType, arrow.Table, error)
+	ToRecordBatches(ctx context.Context) (<-chan arrow.Record, <-chan error, *types.StructType)
```

Comment:
Would be good to add comments on the how this is intended to be used and what the difference to the `ToTable` is.

Reply:
100% I'll add this to the next commit.